services/web/server/src/simcore_service_webserver/resource_manager/settings.py

```diff
@@ -3,39 +3,74 @@
     - config-file schema
     - settings
 """
-from typing import Dict, Optional
+
+from copy import deepcopy
 
 from aiohttp.web import Application
-from models_library.settings.redis import RedisConfig
-from pydantic import BaseSettings, Field, PositiveInt
-from servicelib.aiohttp.application_keys import APP_CONFIG_KEY
+from pydantic import Field, PositiveInt
+from servicelib.aiohttp.application_keys import APP_CONFIG_KEY, APP_SETTINGS_KEY
+from settings_library.base import BaseCustomSettings
+from settings_library.redis import RedisSettings
 
 from .config import CONFIG_SECTION_NAME
 
+CONFIG_SECTION_NAME = "resource_manager"
+
 
-class RedisSection(RedisConfig):
-    enabled: bool = True
+# lock names and format strings
+GUEST_USER_RC_LOCK_FORMAT = f"{__name__}:redlock:garbage_collect_user:{{user_id}}"
 
 
-class ResourceManagerSettings(BaseSettings):
-    enabled: bool = True
+class ResourceManagerSettings(BaseCustomSettings):
 
-    resource_deletion_timeout_seconds: Optional[PositiveInt] = Field(
+    RESOURCE_MANAGER_RESOURCE_TTL_S: PositiveInt = Field(
         900,
         description="Expiration time (or Time to live (TTL) in redis jargon) for a registered resource",
+        # legacy!
+        env=[
+            "RESOURCE_MANAGER_RESOURCE_TTL_S",
+            "WEBSERVER_RESOURCES_DELETION_TIMEOUT_SECONDS",  # legacy
+        ],
     )
-    garbage_collection_interval_seconds: Optional[PositiveInt] = Field(
-        30, description="Waiting time between consecutive runs of the garbage-colector"
+
+    RESOURCE_MANAGER_GARBAGE_COLLECTION_INTERVAL_S: PositiveInt = Field(
+        30,
+        description="Waiting time between consecutive runs of the garbage-colector",
+        # legacy
+        env=[
+            "RESOURCE_MANAGER_GARBAGE_COLLECTION_INTERVAL_S",
+            "WEBSERVER_GARBAGE_COLLECTION_INTERVAL_SECONDS",  # legacy
+        ],
     )
 
-    redis: RedisSection
 
-    class Config:
-        case_sensitive = False
-        env_prefix = "WEBSERVER_"
+def assert_valid_config(app: Application):
+    cfg = app[APP_CONFIG_KEY][CONFIG_SECTION_NAME]
 
+    WEBSERVER_RESOURCE_MANAGER = ResourceManagerSettings()
```

Comment:
pylance does not like these classes with pydantic optionals, did you find a way to circumvent that issue?

Reply:
yes, i noticed. It happened when pydantic was upgraded.

The problem is that pylance notices that there are some argument w/o default in the constructor that are not set. But that is the point of ``BaseSettings`` subclasses, since they can init required fields from envs.

Two options:

- ``type: ignore`` it
- use legacy constructor
```
s = ResourceManagerSettings() # type: ignore
s = ResourceManagerSettings.create_from_envs()
```